extensions/amp-sidebar/1.0/component.js

```diff
@@ -166,11 +167,81 @@ function SidebarWithRef(
         >
           <div className={classes.backdropOverscrollBlocker}></div>
         </div>
-      </>
-    )
+      </div>
+    </>
   );
 }
 
 const Sidebar = forwardRef(SidebarWithRef);
 Sidebar.displayName = 'Sidebar'; // Make findable for tests.
 export {Sidebar};
+
+/**
+ * @param {!SidebarDef.SidebarNavToolbar} props
+ * @return {PreactDef.Renderable}
+ */
+export function SidebarNavToolbar({
+  toolbar,
+  'toolbar-target': toolbarTarget,
+  children,
+  ...rest
+}) {
+  const ref = useRef();
+  const [targetEl, setTargetEl] = useState();
+  const [, setReRender] = useState(true);
+
+  // Trigger re-render each time the window is resized
+  useEffect(() => {
+    const navElement = ref.current;
+    if (!navElement) {
+      return;
+    }
+    const document = navElement.ownerDocument;
+    if (!document) {
+      return;
+    }
+    const window = document.defaultView;
+    if (!window) {
+      return;
+    }
+
+    const reRenderFunction = debounce(
+      window,
+      () => setReRender((prev) => !prev),
+      100
+    );
+    window.addEventListener('resize', reRenderFunction);
+    return () => {
+      window.removeEventListener('resize', reRenderFunction);
+    };
+  }, []);
+
+  useEffect(() => {
+    const navElement = ref.current;
+    if (!navElement) {
+      return;
+    }
+    const document = navElement.ownerDocument;
+    if (!document) {
+      return;
+    }
+    const window = document.defaultView;
+    if (!window) {
+      return;
+    }
+
+    const matches = window.matchMedia(toolbar).matches;
+    const selector = `#${CSS.escape(toolbarTarget)}`;
+    const newTargetEl = document.querySelector(selector);
+    setTargetEl(matches && newTargetEl);
+  });
+
+  return (
+    <>
+      <nav ref={ref} toolbar={toolbar} toolbar-target={toolbarTarget} {...rest}>
+        {children}
+      </nav>
+      {targetEl && createPortal(children, targetEl)}
```

Comment:
Does `children` need to be cloned here?

Reply:
From my understanding it does not, the children are cloned.

Reply:
Nope.